File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode mergeTwoLists(ListNode list1, ListNode list2) {
        return nextNode(list1, list2);
    }

    public ListNode nextNode(ListNode p1, ListNode p2) {
        if (p1 == null && p2 != null) {
            return new ListNode(p2.val, nextNode(p1, p2.next));
        }
        if (p1 != null && p2 == null) {
            return new ListNode(p1.val, nextNode(p1.next, p2));
        }
        if (p1 == null && p2 == null) {
            return null;
        }
        if (p1.val <= p2.val) {
            return new ListNode(p1.val, nextNode(p1.next, p2));
        } else {
            return new ListNode(p2.val, nextNode(p1, p2.next));
        }
    }
}

File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.py
# Definition for singly-linked list.
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

class Solution:
    def mergeTwoLists(self, list1: Optional[ListNode], list2: Optional[ListNode]) -> Optional[ListNode]:
        done = False
        node = self.nextNode(list1, list2)

        return node
    
    def nextNode(self, p1, p2):
        if p1 == None and p2 != None:
            return ListNode(val=p2.val, next=self.nextNode(p1, p2.next))
        elif p2 == None and p1 != None:
            return ListNode(val=p1.val, next=self.nextNode(p1.next, p2))
        elif p1 == None and p2 == None:
            return None
        if p1.val <= p2.val:
            return ListNode(val=p1.val, next=self.nextNode(p1.next, p2))
        else:
            return ListNode(val=p2.val, next=self.nextNode(p1, p2.next))